### PR TITLE
Enhance AST representation of curlies / encapsed strings

### LINKED ISSUE #101: Curly braces in encapsed string, again
When working on prettier support for PHP (#100) i noticed something odd about curly braces in encapsed strings:
```php
"Hello {$foo}";
```
yields
```javascript
[ String {
            kind: 'string',
            loc: 
             Location {
               source: null,
               start: Position { line: 1, column: 1, offset: 1 },
               end: Position { line: 1, column: 7, offset: 7 } },
            value: 'Hello ',
            isDoubleQuote: false },
          Variable {
            kind: 'variable',
            loc: 
             Location {
               source: null,
               start: Position { line: 1, column: 8, offset: 8 },
               end: Position { line: 1, column: 12, offset: 12 } },
            name: 'foo',
            byref: false,
            curly: false } ],
```
But 
```php
"Hello ${foo}";
```
yields 
```javascript
[ String {
            kind: 'string',
            loc: 
             Location {
               source: null,
               start: Position { line: 1, column: 1, offset: 1 },
               end: Position { line: 1, column: 7, offset: 7 } },
            value: 'Hello ',
            isDoubleQuote: false },
          Variable {
            kind: 'variable',
            loc: 
             Location {
               source: null,
               start: Position { line: 1, column: 7, offset: 7 },
               end: Position { line: 1, column: 13, offset: 13 } },
            name: 'foo',
            byref: false,
            curly: true } ],
```
In the first case, the information about the surrounding curly braces seems to be lost, which makes it hard to re-print the node the way it was found. 
What do you think about this? If there is no semantic difference between the two different styles, we might get away with setting `curly:true` in both cases?

### PR DESCRIPTION
**Pull request by czosel**
Fixes #101

### CODE REVIEW

File: src/parser/scalar.js [hunk 1]
```diff
@@ -188,18 +188,17 @@ module.exports = {
       name = null;
       if (this.next().token === this.tok.T_STRING_VARNAME) {
         const varName = this.text();
-        name = this.node("variable");
```
Comment (ichiriac) [inline src/parser/scalar.js]: I need to start the variable node builder before the next statement in order to correctly locate its offset

Comment (czosel) [reply src/parser/scalar.js]: Oh, that explains why it was written that way! I reverted the change.

**Comment by czosel** [comment]
@ichiriac let me know in case anything else needs to be changed :smile:

**Comment by ichiriac** [comment]
It's all good, merged :+1:

**Pull request merged**